jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/MediaFileUpnpProcessor.java

```diff
@@ -62,72 +59,45 @@ public ProcId getProcId() {
         return ProcId.FOLDER;
     }
 
-    private void applyId(MediaFile item, Container container) {
-        container.setId(ProcId.FOLDER.getValue() + ProcId.CID_SEPA + item.getId());
-        container.setTitle(item.getName());
-        container.setChildCount(getChildSizeOf(item));
-        if (mediaFileService.isRoot(item)) {
-            container.setParentID(ProcId.FOLDER.getValue());
-        } else {
-            MediaFile parent = mediaFileService.getParentOf(item);
-            if (parent != null) {
-                container.setParentID(String.valueOf(parent.getId()));
-            }
-        }
+    @Override
+    public Container createContainer(MediaFile entity) {
+        int childSize = getChildSizeOf(entity);
+        return switch (entity.getMediaType()) {
+        case ALBUM -> factory.toAlbum(entity, childSize);
+        case DIRECTORY -> isEmpty(entity.getArtist()) ? factory.toMusicFolder(entity, getProcId(), childSize)
+                : factory.toArtist(entity, childSize);
+        default -> throw new IllegalArgumentException("Unexpected value: " + entity.getMediaType());
+        };
     }
 
     @Override
-    public Container createContainer(MediaFile item) {
-        if (item.isAlbum()) {
-            MusicAlbum container = new MusicAlbum();
-            container.addProperty(factory.toAlbumArt(item));
-            if (item.getArtist() != null) {
-                container.addProperty(factory.toPerson(item.getArtist()));
-            }
-            container.setDescription(item.getComment());
-            applyId(item, container);
-            return container;
-        } else if (item.isDirectory()) {
-            if (isEmpty(item.getArtist())) {
-                StorageFolder container = new StorageFolder();
-                applyId(item, container);
-                return container;
-            }
-            MusicArtist container = new MusicArtist();
-            applyId(item, container);
-            item.getCoverArtPath().ifPresent(path -> container.addProperty(factory.toAlbumArt(item)));
-            return container;
+    public void addItem(DIDLContent parent, MediaFile entity) {
+        if (entity.isFile()) {
+            parent.addItem(factory.toMusicTrack(entity));
+        } else {
+            parent.addContainer(createContainer(entity));
```

Comment:
The name addItem is inappropriate. (The interface is not good) The pattern of adding either Container or Item is often seen.

Reply:
Done.